### PR TITLE
Feedback dialog and form

### PR DESCRIPTION
**Pull request by snurrfint**

### CODE REVIEW

File: test/views/start-page.spec.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import '../../test_bundle/main.test.bundle.js';
+
+describe('start-page', () => {
+  let component;
+
+  beforeEach(() => {
+    component = document.createElement('start-page');
+  });
+  // TODO: Real tests
```
Comment (aihazm) [inline test/views/start-page.spec.js]: :)

File: src/components/dialog/ai-dialog.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
+import AIHTMLElement, { html } from '../ai-html-element';
+import css from './ai-dialog.less';
+
+/**
+   * # Dialog component
+   * <ai-dialog></ai-dialog>
+   *
+   * @property {string} buttontitle - title of the dialog
```
Comment (bjornwiberg) [inline src/components/dialog/ai-dialog.js]: this does not reflect the observed attributes, we observe `dialogtitle`

File: src/components/dialog/ai-dialog.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        this.setOpen(this.isOpen);
+      }
+      this.invalidate();
+    });
+  }
+
+  template() {
+    return html`
+      <style>
+      ${css}
```
Comment (bjornwiberg) [inline src/components/dialog/ai-dialog.js]: indent line `116-120`

File: src/components/dialog/ai-dialog.less [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+      .content {
+        display: flex;
+        flex-direction: column;
+        align-items: center;
+        width: 100%;
+        height: 100%;
+        color: black;
+        border: 1px solid black;
+        transition: @animationSpeed;
+        box-sizing: border-box;
```
Comment (bjornwiberg) [inline src/components/dialog/ai-dialog.less]: separate selectors with newline

File: src/components/dialog/ai-dialog.less [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        span {
+          flex: 0 0 auto;
+          position: relative;
+          font-size: 1.5rem;
+          padding: 0 1em;
+          transition: @animationSpeed;
+          background-color: white;
+          top: -.8em;
+        }
```
Comment (bjornwiberg) [inline src/components/dialog/ai-dialog.less]: separate selectors with newline

File: src/components/dialog/ai-dialog.less [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+          opacity: 0;
+          flex: 1 1 auto;
+          width: 100%;
+          height: 100%;
+          transition: @animationSpeed;
+          transform: translateY(-2em);
+        }
+      }
+    }
+  }
```
Comment (bjornwiberg) [inline src/components/dialog/ai-dialog.less]: separate selectors with newline

File: src/components/dialog/ai-dialog.less [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+  &.open .wrapper .content-wrapper {
+    width: 25em;
+    height: 32em;
+    padding: 2em;
+    bottom: 0;
+    pointer-events: all;
```
Comment (bjornwiberg) [inline src/components/dialog/ai-dialog.less]: separate selectors with newline

File: src/components/dialog/ai-dialog.less [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+button {
+  color: @fontColor;
+  font-family: 'Source Sans Pro';
+  font-size: 1.5rem;
+  padding: .5em 1.5em;
+  box-sizing: border-box;
+  border: 1px solid black;
+  background-color: white;
+  white-space: nowrap;
+  transition: 0@animationSpeed;
```
Comment (bjornwiberg) [inline src/components/dialog/ai-dialog.less]: typo? transition: **0**@animationSpeed;
```suggestion
transition: @animationSpeed;
```

File: src/components/dialog/ai-dialog.less [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+  cursor: pointer;
+
+  &:active {
```
Comment (bjornwiberg) [inline src/components/dialog/ai-dialog.less]:
```suggestion
  &:active,
  &:hover {
    color: @colorGreen;
  }     
```

File: src/components/select/ai-select.less [hunk 10]
```diff
@@ -106,6 +106,7 @@
     position: absolute;
     left: 0;
     top: 3.95em;
+    box-shadow: 0 3px 3px 0 rgba(0,0,0,0.15);
```
Comment (bjornwiberg) [inline src/components/select/ai-select.less]: separate commas with space for readability

File: src/views/start-page/start-page.js [hunk 11]
```diff
@@ -1,11 +1,60 @@
...
 
 customElements.define('start-page', class extends AIHTMLElement {
+  constructor() {
+    super();
+
+    this.resetForm();
+  }
+
+  toggleFeedback() {
+    const feedback = this.root.querySelector('#feedback');
```
Comment (bjornwiberg) [inline src/views/start-page/start-page.js]: separate assignments with newline

File: src/views/start-page/start-page.js [hunk 12]
```diff
@@ -1,11 +1,60 @@
...
+      feedback.removeAttribute('open');
+      this.resetForm();
+    } else {
+      feedback.setAttribute('open', '');
+    }
+  }
+
+  handleFeedback(e) {
+    e.preventDefault();
+    const data = new FormData(e.target);
```
Comment (bjornwiberg) [inline src/views/start-page/start-page.js]: put assignments first and separate with newline

File: src/views/start-page/start-page.js [hunk 13]
```diff
@@ -1,11 +1,60 @@
...
+    /* eslint-enable */
+    this.resetForm();
+    e.target.reset();
+  }
+
+  inputChanged(e, element) {
+    if (element === 'input') {
+      this.feedbackInputValue = e.target.value;
+    } else {
+      this.feedbacktextAreaValue = e.target.value === '' ? null : e.target.value;
```
Comment (bjornwiberg) [inline src/views/start-page/start-page.js]: rename `feedbacktextAreaValue` to `feedbackTextAreaValue` ?

File: src/components/dialog/ai-dialog.less [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+    color: @colorGreen;
+  }
+
+  &:hover {
+    color: @colorGreen;
+  }
+
```
Comment (bjornwiberg) [inline src/components/dialog/ai-dialog.less]: remove extra newline

File: src/components/app.less [hunk 15]
```diff
@@ -87,3 +87,30 @@ p {
...
+button {
+  background-color: white;
+  font-size: 1.2rem;
+  height: 2.5em;
+  padding: .5em 1.5em;
+  color: @fontColor;
+  cursor: pointer;
+  border: 1px solid black;
+
+  &:not(:disabled):active {
```
Comment (bjornwiberg) [inline src/components/app.less]:
```suggestion
  &:not(:disabled) {
    &:active,
    &:hover {
        color: @colorGreen;
    }
  }
```

File: src/components/dialog/ai-dialog.less (hunk 14 above)
Comment (aihazm) [reply src/components/dialog/ai-dialog.less]: maybe consider adding a css/less linter?

File: test/views/start-page.spec.js [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+  it('the feedback dialog should display a form', (done) => {
+    const listener = () => {
+      component.removeEventListener('validated', listener);
+      expect(component.shadowRoot.querySelector('ai-dialog .form') !== null).to.equal(true);
+      expect(component.shadowRoot.querySelector('ai-dialog .happy-container') !== null).to.equal(true);
+      expect(component.shadowRoot.querySelector('ai-dialog .suggestions') !== null).to.equal(true);
+      expect(component.shadowRoot.querySelector('ai-dialog .feedback-button-container') !== null).to.equal(true);
+      done();
+    };
+    component.addEventListener('validated', listener);
```
Comment (bjornwiberg) [inline test/views/start-page.spec.js]: separate assignments with newline

File: test/components/ai-dialog.spec.js [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+    document.documentElement.appendChild(component);
+  });
+
+  it('dialog should be rendered in the DOM', () => {
+    expect(document.querySelector('ai-dialog')).to.equal(component);
+  });
+
+  it('should open dialog on button press', (done) => {
+    const listener = () => {
+      component.removeEventListener('validated', listener);
```
Comment (bjornwiberg) [inline test/components/ai-dialog.spec.js]: put assignments first
```suggestion
      const openButton = component.shadowRoot.querySelector('#open-button');
      const dialog = component.shadowRoot.querySelector('.ai-dialog');
      
      component.removeEventListener('validated', listener);
```

File: src/views/start-page/start-page.less [hunk 18]
```diff
@@ -71,50 +76,139 @@
...
+
+            input {
+              position: absolute;
+              opacity: 0;
+              cursor: pointer;
+              height: 0;
+              width: 0;
+            }
+
+            .happy, .sad {
```
Comment (bjornwiberg) [inline src/views/start-page/start-page.less]: separate selectors with newline for readability
```suggestion
            .happy,
            sad {

File: src/views/start-page/start-page.less [hunk 19]
```diff
@@ -71,50 +76,139 @@
...
+
+            .happy {
+              background-image: url(../../assets/happy_grey.png);
+            }
+
+            .sad {
+              background-image: url(../../assets/sad_grey.png);
+            }
+
+            #happy:focus + .happy, #sad:focus + .sad {
```
Comment (bjornwiberg) [inline src/views/start-page/start-page.less]: separate selectors with newline for readability
```suggestion
            #happy:focus + .happy,
            #sad:focus + .sad {

File: src/views/start-page/start-page.less [hunk 20]
```diff
@@ -17,6 +18,10 @@
       display: flex;
       justify-content: center;
       align-items: center;
+      background-repeat: no-repeat;
```
Comment (bjornwiberg) [inline src/views/start-page/start-page.less]: oneliner?

```suggestion
      background: url("../assets/cubes.png") center/cover no-repeat;
```

**Pull request merged**